pkg/controller/containerjfr/containerjfr_controller.go

```diff
@@ -277,6 +278,25 @@ func (r *ReconcileContainerJFR) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err
 	}
 
+	// Check that secrets mounted in /truststore coincide with CRD
+	err = r.Client.Get(context.Background(), types.NamespacedName{Name: deployment.Name, Namespace: deployment.Namespace}, deployment)
+	if err == nil {
+		deploymentMounts := deployment.Spec.Template.Spec.Containers[0].VolumeMounts
+		expectedMounts := resources.NewDeploymentForCR(instance, serviceSpecs, tlsConfig).Spec.Template.Spec.Containers[0].VolumeMounts
+		if !cmp.Equal(deploymentMounts, expectedMounts) {
+			reqLogger.Info("cert secrets mounted do not coincide with those specified in CRD, modifying deployment")
+			// Modify deployment
+			deployment.Spec.Template.Spec.Containers[0].VolumeMounts = expectedMounts
```

Comment:
Sorry, I overlooked that we need to update the Volumes as well as VolumeMounts here. Otherwise any new mounts we define will be missing the Volumes.